rfc/0000-configuration-and-shared-files.md

```diff
@@ -0,0 +1,296 @@
+  - Unix and Unix-like systems:
+    - `$XDG_CONFIG_HOME/pact/config.toml` (which defaults to `~/.config/pact/config.toml` if `XDG_CONFIG_HOME` is not set)
+  - Windows:
+    - `%APPDATA%\pact\config.toml`
+  - Fallback:
+    - `~/.pact/config.toml`
+
+  > [!NOTE]
+  >
+  > macOS is considered a Unix-like system for the purposes of this RFC. Should an end-user prefer to use a different location (such as `~/Library/Preferences`), they may do so by setting the `XDG_CONFIG_HOME` environment variable.
+
+  If the operating system cannot be detected, the fallback location _should_ be used.
+
+  If the operating system is detected and the configuration file is found in the fallback location, a warning _should_ be displayed informing the user to relocate the configuration. The exception to this is if the location of the config file is explicitly set, in which case the warning _must not_ be displayed.
+
+  If an end user has multiple user configuration files, it is undefined behaviour.
+
+- **Local Configuration File**
+
+  Location:
+
+  - In the process's working directory, either of:
+    - `pact.toml`
+    - `.pact.toml`
+  - Or looking up the directory tree for the first of the above files.
+
+  If both `pact.toml` and `.pact.toml` are found in the same directory, it is undefined behaviour.
+
+  The parsing of the configuration file _must_ stop at the first file found.
+
+#### Format
+
+The TOML configuration file _must_ be organised into sections (or tables) to group configuration data. Top-level configuration data _must not_ be allowed.
+
+The following tables are proposed:
+
+- `pact`
+
+  To store configuration that is shared across the Pact ecosystem:
+
+  ```toml
+  [pact]
+  plugin_dir = "~/.local/share/pact/plugins"
+  data_home = "~/.local/share/pact"
+  do_not_track = true
+  ```
+
+  The following entries are proposed as part of this RFC:
+
+  - `plugin_dir`: The directory where shared plugins are stored. Equivalent to `PACT_PLUGIN_DIR`.
+  - `data_home`: The directory where shared files are stored. Equivalent to `PACT_DATA_HOME`.
+  - `do_not_track`: If set to `true`, the Pact implementation _must not_ report any usage analytics data back to the Pact Foundation.
+
+- `broker`
+
+  To store configuration as to how to connect to the Pact Broker:
+
+  ```toml
+  [broker]
+  base_url = "https://pact-broker.local.company.com:9292"
+  username = "user"
+  password = "password"
+  ```
+
+  The following entries are proposed:
+
+  - `base_url`: The base URL of the Pact Broker. Equivalent to `PACT_BROKER_BASE_URL`.
+  - `username`: The username to use when authenticating with the Pact Broker. Equivalent to `PACT_BROKER_USERNAME`.
+  - `password`: The password to use when authenticating with the Pact Broker. Equivalent to `PACT_BROKER_PASSWORD`.
+  - `token`: The token to use when authenticating with the Pact Broker. Equivalent to `PACT_BROKER_TOKEN`.
```

Comment:
I'm not sure, that if username/password is set then token is ignored and vice versa?

Should we mention that if it's true?

Reply:
When talking about the environment variables, I do state that having a mix of username/password with token is undefined behaviour.

Having said that, this does bring a good question as to how best to handle a local override to a user config; as this could quite easily result in all three being defined.

To resolve this, I actually think it might be best to change the TOML config to:

```toml
[broker]
auth = { token = "..." }
# or
auth = { username = "...", password = "..." }
```